src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Net
+{
+    internal static partial class NameResolutionPal
+    {
+        private static SocketError GetSocketErrorForErrno(int errno)
```

Comment:
Note that these values do not come from `errno`. They'll still need a shim, but it's likely to be networking-specific.